nntrainer/layers/acti_func.cpp

```diff
@@ -39,6 +39,24 @@ int ActiFunc::setActivation(
   std::function<Tensor &(Tensor &, Tensor &, Tensor const &)> const
     &activation_prime_fn) {
   _act_fn = activation_fn;
+  _act_prime_fn =
+    [activation_prime_fn](Tensor const &t_in, Tensor &t_out,
+                          Tensor &outgoing_derivative,
+                          Tensor const &incoming_derivative) -> Tensor & {
+    return activation_prime_fn(t_out, outgoing_derivative, incoming_derivative);
+  };
+
+  return ML_ERROR_NONE;
+}
+
+int ActiFunc::setActivation(
+  std::function<Tensor &(Tensor const &, Tensor &)> const &activation_fn,
+  std::function<Tensor &(Tensor const &, Tensor const &, Tensor &,
+                         Tensor const &)> const &activation_prime_fn) {
+  if (in_place)
```

Comment:
this means that this function is only for the non-inplace ActiFunc?  I'm not sure why we have to set this limitation for this function. It seems to me, something like it removes the opportunity to use this for other ActiFunc for later use.

Reply:
Yes, this function is only for non-inplace activation function. Each of the four parameters of activation_prime_fn means input tensor, output tensor, outgoing derivation, and incoming derivation. In addition, the fact that input and output tensors are independent during backwarding means that they do not support in-place operation(if it supports in-place, then they do not need input and output tensors independently). So, I don't think it makes sense to support the in-place operation here.